test/e2e/branding_test.go

```diff
@@ -0,0 +1,198 @@
+package e2e
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	operatorsv1 "github.com/openshift/api/operator/v1"
+	consoleapi "github.com/openshift/console-operator/pkg/api"
+	"github.com/openshift/console-operator/pkg/testframework"
+)
+
+var validBrands = []operatorsv1.Brand{operatorsv1.BrandOKD, operatorsv1.BrandOCP, operatorsv1.BrandOnline, operatorsv1.BrandDedicated, operatorsv1.BrandAzure}
+
+// Test prep - setup the client used by each test
+func setupBrandingTestCase(t *testing.T) (*testframework.Clientset, operatorsv1.Brand, *v1.ConfigMap) {
+	client := testframework.MustNewClientset(t, nil)
+	// Get the original operator config
+	originalConfig, err := client.Consoles().Get(consoleapi.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("could not get operator config, %v", err)
+	}
+	originalConfigBrand := originalConfig.Spec.Customization.Brand
+	// Get the original Managed Config Map
+	originalManagedConfigMap, err := client.ConfigMaps(consoleapi.OpenShiftConfigManagedNamespace).Get(consoleapi.OpenShiftConsoleConfigMapName, metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		t.Fatalf("could not get console-config configmap, %v", err)
+	}
+
+	return client, originalConfigBrand, originalManagedConfigMap
+}
+
+func cleanupBrandingTestCase(t *testing.T, client *testframework.Clientset, originalConfigBrand operatorsv1.Brand, originalManagedConfigMap *v1.ConfigMap) {
+
+	time.Sleep(1)
+
+	setOperatorConfigBrand(t, client, originalConfigBrand)
+
+	if originalManagedConfigMap.Name != "" {
+		originalManagedConfigMap.SetResourceVersion("0")
+	} else {
+		originalManagedConfigMap = createConsoleConfigMap(operatorsv1.BrandOKD)
+	}
+	err := client.ConfigMaps(consoleapi.OpenShiftConfigManagedNamespace).Delete(consoleapi.OpenShiftConsoleConfigMapName, &metav1.DeleteOptions{})
+	if err != nil {
+		t.Fatalf("could not delete managed config map %s, %v", originalManagedConfigMap.GetName(), err)
+	}
+	_, err = client.ConfigMaps(consoleapi.OpenShiftConfigManagedNamespace).Create(originalManagedConfigMap)
+	if err != nil {
+		t.Fatalf("could not reset managed config map %s, %v", originalManagedConfigMap.GetName(), err)
+	}
+
+}
+
+// Test setting brand using the operator config which overrides the default and managed configmap
+func TestOperatorConfigBranding(t *testing.T) {
+	client, originalConfigBrand, originalManagedConfigMap := setupBrandingTestCase(t)
+	defer cleanupBrandingTestCase(t, client, originalConfigBrand, originalManagedConfigMap)
+	// Set a temporary managed config to test it does not override the operator config values
+	_, err := updateOrCreateConsoleConfigMap(client, consoleapi.OpenShiftConfigManagedNamespace, createConsoleConfigMap(operatorsv1.BrandOKD))
```

Comment:
So the `console-config` in `openshift-config-managed` is responsible for altering the default `brand`, if the configmap exists.  Do we need to retrieve it in the test for the operator config?  This test just needs to verify that changing the value in the operator config does what we expect to the `console-config` in `openshift-console` namespace, correct?

Reply:
This is adding an element to the test that verifies the operator config will have higher priority than the managed config. I could separate these into 2 tests but I think this is clean and proves the point that the operator config wins out.

Reply:
I'm good with this.